Сделал  подписи классов. В index.html

holyflameq opened the pull request.

holyflameq committed:
Сделал  подписи классов. В index.html

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-                  <a href="product.html">
+          <ul class="site-navigation-list">
+            <li class="site-navigation-item"><a href="catalog.html">Каталог</a>
+              <ul class="product-navigation">
```
здесь product не подойдет, так как product - это товар, у тебя же здесь подменю каталога

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-                <li>
-                  <a href="product.html">
+          <ul class="site-navigation-list">
+            <li class="site-navigation-item"><a href="catalog.html">Каталог</a>
```
для стилизации нужно будет обращаться к ссылке, почему ты не задал ей класс? будет вложенность .site-navigation-item a?

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
+            <li class="site-navigation-item"><a href="catalog.html">Каталог</a>
+              <ul class="product-navigation">
+                <li class="product-navigation-item">
+                  <a class="button-product" href="product.html">
```
это разве button?) нежелательно привязывать типы тегов к названиям классов, потому что тег может измениться

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
           </ul>
-        <section>
-          <div>
+        <section class="popap-header">
```
popUp - это всплывающее окно, сам блок который показывается/скрывается, не нужно весь родительский блок именовать как попап.
у секшна должно быть название описывающее в целом, ЧТО ЭТО. у тебя это что?)

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-        <section>
-          <div>
+        <section class="popap-header">
+          <div class="popap-header-search">
```
я вообще против указания местоположения тега (header/main/footer), потому что поиск может так же появиться в футере, например. тогда тебе нужно будет менять полностью класс и все стили вешать на popap-footer-search, а это дублирование кода = замедление загрузки страницы

aendnn replied:
должно быть глобальное название, это не попап. попап то, что выпадает. у тебя пока этого блока в разметке нет (форма поиска).

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
             <input type="search"><!-- Подписи нет. Будет заглушка. Для видимости оставил пока -->
           </div>
-          <div>
+          <div class="popap-header-login">
```
здесь и дальше сделать по аналогии с поиском

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
             <p>Введите пожалуйста свой логин и пароль</p>
             <!-- Здесь будет форма логина -->
-            <button type="submit">Войти</button>
+            <button сlass="login-link" type="submit">Войти</button>
```
здесь нужно переписать форму логина, потому что тут кнопка, а в классе указана ссылка. вернемся к этому после разметки всплывающих форм

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
       </div>
     </header>
-    <main>
+    <main class="container">
```
зачем здесь класс контейнер?)

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
+    <main class="container">
       <h1>Магазин мороженого Gllacy Shop</h1>
-      <section>
+      <section class="big-features">
```
это не преимущества(

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-                <img src="img/studio.jpg" alt="Крем-Брюле">
-                <figcaption>
-                  Крем-брюле и пломбир с малиновым джемом
+        <ul class="big-features-page">
```
page??? почему так? это же список-слайдер

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-                  Крем-брюле и пломбир с малиновым джемом
+        <ul class="big-features-page">
+          <li class="big-features-item1">
+            <a class="button" href="Catalog.html">
```
тег ссылки, зачем ты именуешь как кнопку?) здесь просто переход, тут нет оформления

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-                <figcaption>
-                  Крем-брюле и пломбир с малиновым джемом
+        <ul class="big-features-page">
+          <li class="big-features-item1">
```
у этих элементов будет одинаковая стилизация, если оставить такой класс, тебе нужно будет дублировать .big-features-item1, big-features-item2, big-features-item3, это плохой подход. 
нужно для идентичных элементов делать одинаковые классы, если какой-то из элементов требует доп. стилизации или функционала - добавляешь ему индивидуальный класс

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
+        <ul class="big-features-page">
+          <li class="big-features-item1">
+            <a class="button" href="Catalog.html">
+              <figure class="big-features-item1-image">
```
здесь то же самое: один класс для идентичных элементов, потому что стилизация будет одинаковая

aendnn replied:
это не совсем image, image - картинка внутри, которой ты не задал класс
здесь содержимое слайда, блок-обертка. 
думаю, тут будет уместно слово content

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-          <li><button type="button">Кнопка переключения слайдера</button></li>
-          <li><button type="button">Кнопка переключения слайдера</button></li>
-          <li><button type="button">Кнопка переключения слайдера</button></li>
+          <li><button class="slider" type="button">Кнопка переключения слайдера</button></li>
```
slider - слишком глобальный класс, он больше подходит для секции, в которой эти переключалки находятся

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
         </ul>
       </section>
-      <section>
+      <section class="medium-features">
```
на странице обычно больше 1 блока с преимуществами не бывает. тут тоже не преимущества( советую переводить заголовки в переводчике, так будет проще именовать секции поначалу

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
+      <section class="medium-features">
         <h2>Акции и суперпредложения!</h2>
-        <div>
+        <div class="medium-features-item1">
```
про item1/item2 писала выше

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-          <p>При покупке 2 кг любого фруктового мороженого, добавим в ваш заказ банку малинового варенья бесплатно.</p>
-            <a href="catalog.html">Хочу варенье!</a> <!-- Мне кажется тут кнопка а не ссылка. Кнопка типа <button type="submit">Хочу варенье!</button> -->
+          <p class="medium-features-item1-text">При покупке 2 кг любого фруктового мороженого, добавим в ваш заказ банку малинового варенья бесплатно.</p>
+            <a class="button" href="catalog.html">Хочу варенье!</a> <!-- Мне кажется тут кнопка а не ссылка. Кнопка типа <button type="submit">Хочу варенье!</button> -->
```
ссылке нужно добавить доп. класс для позиционирования

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
+                <span class="catalog-hit-text">Хит!</span> <!-- Пока классов у нас не было. Я хочу оставить span. чтобы не путаться-->
                 <img src="img/orange.jpg" alt="Сливочное с апельсиновым джемом и цитрусовой стружкой">
-                <p><b>310 ₽</b>/кг</p>
+                <p class="catalog-hit-text-price"><b>310 ₽</b>/кг</p>
```
text лишнее

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-                <span>Хит!</span> <!-- Пока классов у нас не было. Я хочу оставить span. чтобы не путаться-->
+        <ul class="hits-list">
+          <li class="hits-item"><a href="Catalog.html">
+              <div class="catalog-hit-image">
```
у тебя выше нет каталога, только hits. для единообразия, сделай в одном стиле: hit-image/ hit-item-image

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
               </div>
               <h3>Сливочное с апельсиновым <br>джемом и цитрусовой стружкой</h3>
             </a>
             <button type="button">Быстрый просмотр</button>
```
доп.класс для стилизации и позиционирования нужен

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
         </ul>
       </section>
-      <section>
+      <section class="advantages">
```
здесь всё идеально, молодец! и лист и айтем, всё супер. 
этот блок - это те самые features, которых у тебя аж 2 штуки выше)

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
-          <p>Новое в нашем блоге</p>
-          <a href="blog.html"><b>10 способов сервировки <br>фруктовых щербетов <br>к столу</b></a>
+          <p class="blog-text-caption">Новое в нашем блоге</p>
+          <a class="blog-text" href="blog.html"><b>10 способов сервировки <br>фруктовых щербетов <br>к столу</b></a>
```
я считаю, что это заголовок статьи, а ты?)

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
+          <a class="blog-text" href="blog.html"><b>10 способов сервировки <br>фруктовых щербетов <br>к столу</b></a>
         </article>
-        <section>
+        <section class="news-letter">
```
тут не нужен дефис

aendnn commented on index.html:
```diff
@@ -1,195 +1,195 @@
           <h2>Подписка на рассылку</h2> <!-- просто показал заголовок-->
-          <p>Подпишитесь на нашу сладкую рассылку и будьте всегда в курсе <br>всего самого вкусного, что у нас происходит. Обещаем не спамить <br>и не слать всякой ненужной ерунды. Честно =)</p>
+          <p class="news-letter-text">Подпишитесь на нашу сладкую рассылку и будьте всегда в курсе <br>всего самого вкусного, что у нас происходит. Обещаем не спамить <br>и не слать всякой ненужной ерунды. Честно =)</p>
           <form action="https://Glacy" method="post">
```
почему у формы нет класса? и во внутреннем диве он неправильный, это обертка

aendnn commented on index.html:
```diff
@@ -200,14 +200,14 @@ <h2>Подписка на рассылку</h2> <!-- просто показал
         <h2>Как до нас добраться? Контактная информация</h2>
         <div>
-          <p>
+          <p class="contacts-text-adress">
```
обозначение text лишнее

aendnn commented on index.html:
```diff
@@ -216,44 +216,44 @@ <h2>Как до нас добраться? Контактная информац
     </main>
-    <footer>
-      <section>
+    <footer class="page-footer main-footer">
```
а зачем тут 2 класса?) ответ: "так в демке" не принимается

aendnn commented on index.html:
```diff
@@ -216,44 +216,44 @@ <h2>Как до нас добраться? Контактная информац
-          <li>
-            <a href="#">Фейсбук</a>
+        <ul class="footer-social-list">
+          <li class="footer-social-item-f">
```
про идентичные элементы писала выше

aendnn commented on index.html:
```diff
@@ -216,44 +216,44 @@ <h2>Как до нас добраться? Контактная информац
-          <li>
-            <a href="#"><b>Для поставщиков</b></a>
+        <ul class="footer-contacts-list">
+          <li class="footer-contacts-item-left">
```
здесь идентичные элементы, некоторые из них отличаются, значит нужен доп. класс
позиционирование тут ставится автоматом, указывать какой где элемент не требуется

holyflameq committed:
Cделал правки к Index

aendnn commented on index.html:
```diff
@@ -1,264 +1,264 @@
+        <ul class="big-offers-list-slider">
+          <li class="big-offers-item">
+            <a class="big-offers-link" href="Catalog.html">
+              <figure class="big-offers-item-slide-content">
```
слишком длинный получился класс, поменяй как обсуждали в тг

aendnn commented on index.html:
```diff
@@ -1,264 +1,264 @@
-                <span>Хит!</span> <!-- Пока классов у нас не было. Я хочу оставить span. чтобы не путаться-->
+        <ul class="hits-list">
+          <li class="hits-item"><a class="hits-item-link" href="Catalog.html">
+              <div class="hits-item-image">
```
здесь не только image, еще и прайс

aendnn commented on index.html:
```diff
@@ -1,264 +1,264 @@
-          <li>
-            <a href="#"><b>Для поставщиков</b></a>
+        <ul class="footer-contacts-list">
+          <li class="footer-contacts-item footer-contacts-item-left">
```
доп. классы лишние, можешь оставить конечно, но ты поймешь потом, что по ним не будет никакой стилизации)

aendnn commented on index.html:
```diff
@@ -1,264 +1,264 @@
+      <section class="footer-copyright">
         <h2>HTML Academy</h2>
-        <a href="https://htmlacademy.ru/intensive/htmlcss">
+        <a class="button footer-copyright-link" href="https://htmlacademy.ru/intensive/htmlcss">
```
ты сделал по аналогии с барбершопом, но в нем присутствует класc button потому что в макете ссылка сделана как кнопка, как все кнопки в том макете. у тебя же просто ссылка, внутри которой лого и все

holyflameq committed:
сделал правки в каталоге.

holyflameq committed:
Правки к каталогу

aendnn commented on catalog.html:
```diff
@@ -54,47 +54,47 @@ <h2>Личный кабинет</h2>
-                  </td class="cart-content-item-price-full">
-                  <td>
+                  </td>
+                  <td class="cart-item-price-count">
```
count где кол-во и стоимость

holyflameq committed:
Правки в индекс. и каталог

holyflameq committed:
Индекс и каталог

Не задал кнопкам их состояния.
при наведении, без наведения и тд.
задал, как смог стили для классов

aendnn reviewed (changes requested):
правки

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  font-weight: 400;
+  src: local('Roboto'),
+    url('../fonts/roboto-v30-latin-regular.woff2') format('woff2'),
+    /* Chrome 26+, Opera 23+, Firefox 39+ */
```
Подключать шрифты нужно учиться самостоятельно

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
+}
+
+/* roboto-500 - latin */
```
Для русского текста нужна кириллица

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  /* index-slider */
+  --dark-green: #849D8F;
+  --blue-gray: #8996A6;
+  --brown-light: #9D8B84;
```
Правильнее оттенок задавать перед цветом, как ты сделал с dark чуть выше

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+
+body {
+  mid-width: 1200px;
+  margin: 0px;
```
Если у тебя подключён нормалайз, то обнуление для body уже прописано, дублировать не нужно

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  mid-width: 1200px;
+  margin: 0px;
+  padding: 0px;
+  font-family: 'Roboto';
```
Нужно прописать запасной шрифт на случай если твой не подключится. + если даже запасного нет - прописать какой шрифт нужно подключить serif(с засечками) или sans-serif(без засечек), смотря какой у тебя основной шрифт (с засечками или без)

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  font-family: 'Roboto';
+  font-weight: 700;
+  background-color: var(--dark-green);
+  background-repeat: no-repeat;
```
Фон повторяется только если есть фоновое изображение, у тебя только фортовый цвет

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  background-color: var(--dark-green);
+  background-repeat: no-repeat;
+  color: var(--basic-white);
+}
```
Обычно в body прописываются ещё размер шрифта и высота строки основного текста, чтобы не прописывать потом каждому абзацу размеры. посмотри по макету какой размер чаще всего повторяется и пропиши его тут

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  height: auto;
+}
+
+.site-navigation-list {
```
По умолчанию у списков есть отступы и внутренние и внешние, нужно обнулить margin, padding и прописать list-style-type: none. Должны попасть кружочки

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+.button {
+  background-image: linear-gradient(180deg, #F26843 0%, #E74A35 100%);
+  box-shadow: 0px 2px 2px rgba(172, 16, 0, 0.64);
+  font: inherit;
```
В нормалайзе прописано ужр

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+
+.promotions-item {
+  width: 560px;
+  height: 229px;
```
не рекомендуется задавать высоту таким элементам, если текста станет больше, он вылезет за пределы блока, а он должен растягиваться по высоте, если текста становится больше

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  color: var(--full-black);
+}
+
+.advantages h3 {
```
Нет класса для заголовка? Почему тег стилизуешь?

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  width: 560px;
+  height: 229px;
+}
+
```
Почему не по классу?)

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  font-weight: 400;
+  font-size: 13px;
+  line-height: 18px;
+  border-bottom: 1px solid rgba(255, 255, 255, 0.2);
```
Тут нужно чтобы ты скинул скрин с макета и скрин того, что у тебя получилось

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  border-bottom: 1px solid rgba(255, 255, 255, 0.2);
+}
+
+.footer-copyright p {
```
Здесь нет класса? Это не ошибки, где стилизация по тегу, мне просто интересно почему так, ты же вроде как давал каждому элементу класс

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+}
+
+.footer-copyright-link {
+  border-bottom: 1px solid rgba(255, 255, 255, 0.2);
```
Тут нужно чтобы ты скинул скрин с макета и скрин того, что у тебя получилось

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+}
+
+/* catalog */
+.page-body {
```
У тебя класс page-body и у каталога и у главной. Ты в самом начале стилизовал уже этот класс, эти стили применятся ко всем элементам, у которых есть этот класс, так что дублировать те же стили не нужно

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+
+/* catalog */
+.page-body {
+  mid-width: 1200px;
```
Мид :)

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+.site-navigation-button {
+  background-image: linear-gradient(180deg, #F26843 0%, #E74A35 100%);
+  box-shadow: 0px 2px 2px rgba(172, 16, 0, 0.64);
+  font: inherit;
```
Необязательно дублировать так как есть в нормалайзе

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+  border-radius: 70px;
+}
+
+.site-navigation-item {
```
Лучше стили для меню вынести в начало файла, после стилей хедера, например. Потому что они относятся ко всем страницам сайта, не только к каталогу

aendnn commented on css/style.css:
```diff
@@ -1 +1,348 @@
+}
+
+.catalog-list {
+  list-style: none;
```
Везде где есть списки, сделать как я писала выше

holyflameq committed:
пушу правки. кнопке состояния не задавал

holyflameq committed:
это пример задания для заголовка.

Знаю что неверно

aendnn commented on css/style.css:
```diff
@@ -1 +1,331 @@
+
+body {
+  mid-width: 1200px;
+  font-family: 'Roboto', 'Arial', sans-serif;
```
Ну какой ты умничка 😍

aendnn commented on css/style.css:
```diff
@@ -1 +1,331 @@
+}
+
+body {
+  mid-width: 1200px;
```
мид опять)

aendnn commented on css/style.css:
```diff
@@ -1 +1,331 @@
+.site-navigation-list {
+  line-height: 18px;
+  list-style: none;
+  margin: 0px;
```
если значение 0, то можно убрать px

aendnn commented on css/style.css:
```diff
@@ -1 +1,331 @@
+  padding: 0px;
+}
+
+.page-body a {
```
Класс бади лишний, у тебя на каждой странице он, можно просто глобально задать тегу а стилизацию. В критерии Д25 есть исключения для бади, а и имг

aendnn commented on css/style.css:
```diff
@@ -1 +1,331 @@
+  line-height: 24px;
+}
+
+h1 {
```
не пройдёшь по критерию Д25,это глобальный тег и он не разрешён(

holyflameq replied:
удалил его нахуй) ахъааххаахахахах

aendnn commented on css/style.css:
```diff
@@ -1 +1,331 @@
+  line-height: 24px;
+}
+
+.footer-social-list {
```
Можно задать эти стили для всех списков в футере, чтобы не дублировать одни и те же стили для разных классов

holyflameq replied:
не пон

aendnn replied:
footer-social-list, 
footer-contacts-list одинаковые стили сброса. Почему бы не прописать .footer ul и обнулить 2 списка?)

aendnn replied:
Не сделал, ок) твоё решение

aendnn commented on css/style.css:
```diff
@@ -1 +1,331 @@
+}
+
+.article-text b {
+  text-decoration: underline;
```
Ты спрашивал почему у тебя не такая граница, как в макете. Потому что так работает подчёркивание, вместо него нужно задать бордер

aendnn commented on css/style.css:
```diff
@@ -1 +1,331 @@
+  border-radius: 70px;
+}
+
+.site-navigation-item {
```
Лучше стили для меню и его элементов  вынести в начало файла, после стилей хедера, например. Потому что они относятся ко всем страницам сайта, не только к каталогу

holyflameq replied:
не пон

aendnn replied:
почему у тебя сайт навигейшн почти в конце файла стилей?

holyflameq committed:
Сделал правки. приступаю к кнопке

holyflameq committed:
Сделал некоторый стайл-гайд. Остались вопросы

Написал формы.
осталась форма для отзыва.
+ остался стайлгайд для ссылок.
Наведение-ховер
Нажатие-актив
а стоячее состояние. Дефолтное - это что?

holyflameq committed:
Внес правки. осталась косметика

Сделал. Всё, что мог. Осталось косметику сделать. Классы подписать. Задать цвета для форм.
это завтра- я устал

aendnn commented on css/style.css:
```diff
@@ -1 +1,414 @@
+body {
+  min-width: 1200px;
+  font-family: 'Roboto', 'Arial', sans-serif;
+  font-weight: 700;
```
В макете большинство текста жирный? И советую свойства для шрифта задавать рядышком, жирность/размер/высота линии. Так код будет аккуратнее

holyflameq replied:
да жирный

aendnn commented on css/style.css:
```diff
@@ -1 +1,414 @@
+  border-bottom: 1px solid rgba(255, 255, 255, 0.2);
+}
+
+.site-navigation-list {
```
Этот список нужно перенести выше в коде, чем его айтемы. 
Сначала родитель - потом его потомки

aendnn commented on css/style.css:
```diff
@@ -1 +1,414 @@
+  color: var(--basic-white)
+}
+
+.big-offers-link:hover {
```
Состояния элемента прописываются сразу после его стилизации. 
Чтобы не листать вниз весь код и не искать состояния, они должны быть рядом. 
Элемент -> ниже его состояния

aendnn commented on css/style.css:
```diff
@@ -1 +1,414 @@
+}
+
+/* catalog */
+.user-navigation-cart-link {
```
Это не элемент каталога, это элемент всего хедера.

holyflameq replied:
Cмотри. Меня смутило. что эта ссылка. Именно когда 2 товара в корзине- только в каталоге.
для Индекса я задал другую ссылку с 0 товаров

aendnn replied:
Это ссылка на полноценную корзину и там и там. По наведению - всплывашка с товарами, по клику - переход на полноценную страницу с корзиной, я так это вижу

aendnn reviewed (changes requested):
Все отлично, кроме того, что ты путаешь пока в каком месте прописывать элементы. 
Пару правил:
1. Все, что внутри хедера - прописываешь после глобальных тегов 
2. Состояния элементов должны быть рядом с самим элементом, не в конце файла 
3. Навигация и ее внутренние элементы должны быть в одном месте, нельзя перемещать элементы в разные места в коде, даже если тебе кажется, что это элемент другой страницы

holyflameq committed:
Пушу все с правками

aendnn reviewed (changes requested):
Все отлично, графику с телефона проверить нереально, она останется на потом. В остальном все правильно, элементы по файлу правильно размещены и подписаны комментами, это хорошая практика, ты молодец 😽
Делай дальше блоки
